### PR TITLE
NXP-26305: add NXQL NOW() with optional added period (10.10)

### PR DESCRIPTION
**Pull request by efge**

### CODE REVIEW

File: nuxeo-common/src/main/java/org/nuxeo/common/utils/PeriodAndDuration.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+    protected static int parseFraction(String string, int sign, String text) {
+        if (string == null || string.length() == 0) {
+            return 0;
+        }
+        try {
+            string = (string + "000000000").substring(0, 9);
+            return Integer.parseInt(string) * sign;
+        } catch (NumberFormatException | ArithmeticException e) {
+            throw (DateTimeParseException) new DateTimeParseException(
```
Comment (kevinleturc) [inline nuxeo-common/src/main/java/org/nuxeo/common/utils/PeriodAndDuration.java]: I'm probably missing something, why not using the constructor taking a Throwable?

Comment (efge) [reply nuxeo-common/src/main/java/org/nuxeo/common/utils/PeriodAndDuration.java]: Indeed, I was just using the same code as `java.time.Duration`. I'll fix.

**Comment by efge** [comment]
10.10 T&P https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-fguillaume-10.10/26/

File: nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/NXQL.java [hunk 2]
```diff
@@ -18,13 +18,29 @@
  */
 package org.nuxeo.ecm.core.query.sql;
 
+import java.time.format.DateTimeParseException;
+
+import org.joda.time.DateTime;
+import org.joda.time.chrono.ISOChronology;
```
Comment (ataillefer) [inline nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/NXQL.java]: Didn't we want to get rid of `joda.time`? See https://jira.nuxeo.com/browse/NXP-26517.

File: nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java [hunk 3]
```diff
@@ -45,4 +48,19 @@ public void testSimpleXPath() throws Exception {
         assertEquals("abc/def1", simpleXPath("abc/def1"));
     }
 
+    @Test
+    public void testNowPeriodAndDuration() throws Exception {
+        DateTime dateTime = new DateTime(2001, 2, 3, 4, 5, 6, 7, ISOChronology.getInstanceUTC());
+        assertEquals("2001-02-03T04:05:06.007Z", dateTime.toString());
+        //
```
Comment (ataillefer) [inline nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java]: ?

File: nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/NXQL.java (hunk 2 above)
Comment (efge) [reply nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/NXQL.java]: Yes on master I'm replacing everything with `ZonedDateTime`

File: nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java (hunk 3 above)
Comment (efge) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java]: Just spacing before the actual tests

File: nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/NXQL.java (hunk 2 above)
Comment (efge) [reply nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/core/query/sql/NXQL.java]: See #3182

File: nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java (hunk 3 above)
Comment (ataillefer) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java]: Yep, but why do you need the `//`?

Comment (efge) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/test/java/org/nuxeo/ecm/core/storage/sql/TestNXQLQueryMaker.java]: A blank line was too much whitespace in my eye 😀

**Pull request merged**